### ISSUE TITLE
provideQueryClientOptions changes behaviour of queries even when not changing any default properties

### ISSUE THREAD

**Issue by stevebrowndotco** [ngneat/query#107]
### Which @ngneat/query-* package(s) are the source of the bug?

query

### Is this a regression?

Yes

### Description

If I use `provideQueryClientOptions`, and pass in an empty object, it changes the behaviour of the queries when it shouldn't.

Specifically, one of my queries is stuck in the "fetching" state

// app.module.ts
```
  provideQueryClientOptions({
      defaultOptions: {}
    })
```

### Please provide the exception or error you saw

```true
My query is stuck in "fetching"
```

### Do you want to create a pull request?

No

**Comment by luii** [comment]
`@ngneat/query` is just a mere wrapper/adapter to the underlying `@tanstack/query`, it doesnt transform any default options, every default option that is set from the get go is set by `@tanstack/query`.

Taken from the actual file ([packages/ng-query/src/lib/providers.ts](https://github.com/ngneat/query/blob/main/packages/ng-query/src/lib/providers.ts#L4-L12)):
```ts
export const QUERY_CLIENT_OPTIONS = new InjectionToken<QueryClientConfig>(
  'QUERY_CLIENT_OPTIONS',
  {
    providedIn: 'root',
    factory() {
      return {}; // <-- 
    },
  }
);
```

as you can see it only provides an empty object, that later gets passed to the QueryClient Provider ([packages/ng-query/src/lib/query-client.ts](https://github.com/ngneat/query/blob/main/packages/ng-query/src/lib/query-client.ts#L13-L18)):
```ts
const QueryClient = new InjectionToken<QueryCore>('QueryClient', {
  providedIn: 'root',
  factory() {
    return new QueryCore(inject(QUERY_CLIENT_OPTIONS));
  },
});
```
these are the only references and it doesnt get transformed midway.
I think that your "bug" is not related to either of these packages. For the "stuck in fetching" part we would need to see an example, a stackblitz or reproducable repository would be good.

@NetanelBasal i think you'd agree if i say that this issue can be closed and continued in the discussions section

**Comment by stevebrowndotco** [comment]
Yes I don't mind if you move to the discussions page. I understand that this is an adapter but was wondering if the behaviour is still introduced on this side. I also use tanstack query in react projects and don't get the same problem unfortunately
